pkg/webhook/cloudprovider/ensurer.go

```diff
@@ -0,0 +1,70 @@
+// Copyright (c) 2023 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cloudprovider
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/gardener/gardener-extension-provider-aws/pkg/aws"
+	"github.com/gardener/gardener/extensions/pkg/webhook/cloudprovider"
+	gcontext "github.com/gardener/gardener/extensions/pkg/webhook/context"
+
+	"github.com/go-logr/logr"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// NewEnsurer creates cloudprovider ensurer.
+func NewEnsurer(logger logr.Logger) cloudprovider.Ensurer {
+	return &ensurer{
+		logger: logger,
+	}
+}
+
+type ensurer struct {
+	logger logr.Logger
+	client client.Client
+}
+
+// InjectClient injects the given client into the ensurer.
+func (e *ensurer) InjectClient(client client.Client) error {
+	e.client = client
+	return nil
+}
+
+// InjectScheme injects the given scheme into the decoder of the ensurer.
+func (e *ensurer) InjectScheme(_ *runtime.Scheme) error {
+	return nil
+}
+
+// EnsureCloudProviderSecret ensures that cloudprovider secret contains
+// the shared credentials file.
+func (e *ensurer) EnsureCloudProviderSecret(ctx context.Context, _ gcontext.GardenContext, new, _ *corev1.Secret) error {
+	if _, ok := new.Data[aws.AccessKeyID]; !ok {
+		return fmt.Errorf("could not mutate cloudprovider secret as %q field is missing", aws.AccessKeyID)
+	}
+	if _, ok := new.Data[aws.SecretAccessKey]; !ok {
+		return fmt.Errorf("could not mutate cloudprovider secret as %q field is missing", aws.SecretAccessKey)
+	}
```

Comment:
I am assuming that means you went with @dimityrmirchev first suggestion to not allow users to specify their own credentials file so that you always overwrite it right ?

Reply:
Yes, users will not be able to specify credential files and the secret will remain the same. If you think it's better I can make it the other way and change the validating web hook as well.

Reply:
Now I am wondering if we always overwrite it do we really need the `useCredentialsFile` [boolean](https://github.com/gardener/gardener-extension-provider-aws/pull/683/files#diff-6a010d50f9d04c84f1da3b131822d1b10d39acac136bc54ab0b32b68a3698eaeR24)? I guess this will always be true

Reply:
We would still need `useCredentialsFile` since our webhook would not mutate the `cloudprovider` secret unless it has been updated by the user.

Reply:
After a joint review with @dimityrmirchev we discovered that in fact `useCredentialsFile` is not needed. During shoot reconciliation the `cloudprovider` secret is updated and the new `credentialsFile` field is always added.